src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,554 @@
+                    received_asset_ids: itemsToReceive || [],
+                    other_steam_id64: offer.partner.getSteamID64(),
+                });
+            } catch (error) {
+                console.error(`failed to annotate offer ${offer.id} post-hoc`, error);
+            }
+        }
+    }
+
+    private async pingCancelTrades(
+        tradesInPending: ITradeFloat[],
+        tradeOffers: IGetTradeOffersResponse
+    ): Promise<void> {
+        const hasWaitForCancelPing = tradesInPending.find(
+            (trade) =>
+                trade.state === EStatusTradeCSFLOAT.PENDING &&
+                trade.wait_for_cancel_ping
+        );
+
+        if (!hasWaitForCancelPing) return;
+
+        const allTradeOffers = [
+            ...(tradeOffers.sent || []),
+            ...(tradeOffers.received || []),
+        ];
+
+        for (const trade of tradesInPending) {
+            if (trade.state !== EStatusTradeCSFLOAT.PENDING) continue;
+
+            if (!trade.wait_for_cancel_ping) continue;
+
+            const tradeOffer = allTradeOffers.find(
+                (steamTradeOffer) => steamTradeOffer.id === trade.steam_offer.id
+            );
+
+            if (
+                tradeOffer &&
+                (tradeOffer.state === ETradeOfferStateCSFloat.Active ||
+                    tradeOffer.state === ETradeOfferStateCSFloat.Accepted)
+            )
+                continue;
+
+            try {
+                await this._csFloatClient.pingCancelTrade({
+                    trade_id: trade.id,
+                    steam_id: this.steamIDBase64,
+                });
+            } catch (error) {
+                console.error(
+                    `failed to send cancel ping for trade ${trade.id}`,
+                    error
+                );
+            }
+        }
+    }
+
+    private async autoAcceptTrades(
+        tradesInQueue: ITradeFloat[],
+        tradeOffers: IGetTradeOffersResponse
+    ): Promise<void> {
+        if (tradesInQueue.length < 1) return;
+
+        const tradeOffersSent = tradeOffers.sent;
+
+        for (const tradeInQueue of tradesInQueue) {
+            // const hasMatchingItem = itemsTradables.some(
+            //     (item: CEconItem) => item.assetid === tradeInQueue.contract.item.asset_id
+            // );
+
+            const hasMatchingItem = await this.inInventory(720,2, tradeInQueue.contract.item.asset_id)
+
+            if (!hasMatchingItem) continue;
+            let alreadyHasOffer = false
+            let state: ETradeOfferStateCSFloat;
+
+            for (const tradeOffer of tradeOffersSent) {
+                state = tradeOffer.state
+                for (const item of tradeOffer.itemsToGive) {
+                    if (item.assetid === tradeInQueue.contract.item.asset_id && state != ETradeOfferStateCSFloat.Declined && state != ETradeOfferStateCSFloat.Canceled) {
+                        alreadyHasOffer = true
+                    }
+                }
+            }
```

Comment:
Wrong var scope. Also you can break out of the loop since the verification is already done
```suggestion
            if (!hasMatchingItem) continue;
            let alreadyHasOffer = false

            for (const tradeOffer of tradeOffersSent) {
                const state = tradeOffer.state
                for (const item of tradeOffer.itemsToGive) {
                    if (item.assetid === tradeInQueue.contract.item.asset_id && state != ETradeOfferStateCSFloat.Declined && state != ETradeOfferStateCSFloat.Canceled) {
                        alreadyHasOffer = true
                        break;
                    }
                }
                if (alreadyHasOffer) break;
            }
```